src/components/NewStack.js

```diff
@@ -126,24 +127,36 @@ class NewStack extends PureComponent {
 
   onSubmit (e) {
     e.preventDefault()
+    console.log(e)
 
     let updateOperationsState = false
 
-    const operations = this.state.operations.map((operation) => {
-      const required = this.props.operations[operation.name].required || []
-      const addedOptions = Object.keys(operation.options)
-      const missing = required.filter((field) => {
-        const index = addedOptions.indexOf(field)
-        return index === -1 || !operation.options[field].length
-      })
+    const ajv = new Ajv({
+      allErrors: true
+    })
 
-      operation.errors = {}
-      missing.forEach((field) => {
-        operation.errors[field] = `${field} is required`
-      })
+    this.operationValidators = {}
+    Object.keys(this.props.operations).forEach(key => {
+      const operation = this.props.operations[key]
+      if (key === 'noop' && Array.isArray(operation.properties)) {
+        operation.properties = {}
+      }
+      this.operationValidators[key] = ajv.compile(operation)
+    })
```

Comment:
can you try to move this to the constructor? or maybe you need to move it into component will receive props because it might be that when you open the NewStack page form the very beginning on, the `this.props.operations` is not yet filled because the AJAX request has not been finished yet. 

Also can you do the same for the stackoptions? You should have the stack options properties in `this.props.stackOptions` as far as I know.